test/integration/array_test.ts

```diff
@@ -558,6 +558,35 @@ describe('Array', function () {
     }, this.test!.title);
   });
 
+  it('Can handle concurrent concurrent delete operations', async function() {
```

Comment:
Probably we can also add `length` check in the below two tests 
`Can handle concurrent insertBefore and delete operations`
`Can handle complex concurrent insertBefore and delete operations`. How about this?

Reply:
Great! I will add the assertion to these tests.